### PR TITLE
Update heading height to match figma 🚀

### PR DESCRIPTION
**Pull request by alycecristines**
[JIRA Issue](https://gympass.atlassian.net/browse/EUEG-1765)

## Description 📄

This PR adjust some stylization params to match figma prototype for the heading component and allow the component to receive children as optional.

## Platforms 📲

- [x] Web
- [ ] Mobile

## Type of change 🔍

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? 🧪

[Enter the tips to test this PR]

- [ ] Unit Test
- [x] Snapshot Test

## Checklist: 🔍

- [x] My code follows the contribution guide of this project [Contributing Guide](https://github.com/Gympass/yoga/blob/master/CONTRIBUTING.md)
- [x] Layout matches design prototype: [FIGMA](https://www.figma.com/file/RbIbvb1aLcrKe3fw3ZCjQT/%E2%9C%85-Yoga-%5BNATIVE%5D?node-id=1%3A13815)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Screenshots 📸

[Upload your screenshots here]

| Before | After |
| ------ | ----- |
|       |       |
|       |       |

### CODE REVIEW

File: packages/yoga/src/Heading/Heading.theme.js [hunk 1]
```diff
@@ -3,11 +3,13 @@ const Heading = ({ spacing, colors }) => ({
   padding: {
     vertical: spacing.small,
     horizontal: spacing.huge,
+    horizontalMobile: spacing.large,
```
Comment (leonardokl) [inline packages/yoga/src/Heading/Heading.theme.js:6]: The prototype have `20px` of horizontal padding.

```suggestion
    horizontalMobile: spacing.medium,
```

File: packages/yoga/src/Heading/web/RightButton.jsx [hunk 2]
```diff
@@ -1,9 +1,29 @@
 import React from 'react';
+import styled from 'styled-components';
 import PropTypes from 'prop-types';
 import Button from '../../Button';
 
+const ButtonIcon = styled(Button.Icon)`
+  :not(:last-of-type) {
+    margin-right: 20px;
```
Comment (leonardokl) [inline packages/yoga/src/Heading/web/RightButton.jsx:8]: This should be a token

File: packages/yoga/src/Heading/Heading.theme.js [hunk 3]
```diff
@@ -3,11 +3,13 @@ const Heading = ({ spacing, colors }) => ({
...
   },
-  height: 48,
+  height: 56,
   button: {
-    width: 48,
-    height: spacing.xxlarge,
+    width: spacing.large,
+    height: spacing.large,
+    marginLeft: spacing.medium,
```
Comment (leonardokl) [inline packages/yoga/src/Heading/Heading.theme.js:12]: I think this token is not being used

**Review by leonardokl — APPROVED** [review]

**Review by ruanblimaig — APPROVED** [review]

**Review by ericcleao — APPROVED** [review]

**Comment by ericcleao** [comment]
Just a doubt about the `Heading ButtonIcon`, its touchable area is not too small? Should it have padding or touchable opacity to increase it at least by 10px to each side (assuming the 10px because of the distance between buttons of 20px, but maybe 8px is already fine to increase the area for touchable devices and the fact that we have space to do this at the top and bottom of the component itself)?

**Review by guilhermecardoso-gympass — APPROVED** [review]

**Pull request merged**